### ISSUE TITLE
Responsive .btn-block

### ISSUE THREAD

**Issue by peiche** [twbs/bootstrap#20617]
I love the responsive float classes in v4, but I was wondering if a similar concept could be applied to the `.btn-block` class. Here is a demo page which, as an example, has paging buttons. Without the convenience of a `.btn-[size]-block` class, you end up duplicating markup. (I only made the smallest size for the sake of the example.)

http://codepen.io/peiche/pen/wzvEyJ

**Comment by twbs-lmvtfy** [comment]
Hi @peiche!

You appear to have posted a live example (http://codepen.io/peiche/pen/wzvEyJ.html), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 10, column 3: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.
* line 11, column 3: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.
* line 12, column 3: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.
* line 13, column 3: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.
* line 18, column 3: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.
* line 19, column 3: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by peiche** [comment]
The errors are fixed.
http://codepen.io/peiche/pen/wzvEyJ

**Comment by cvrebert** [comment]
Duplicate of #13647.

**Comment by vanduynslagerp** [comment]
The tricky part would be the vertical spacing between `btn-{breakpoint}-block` [here](https://github.com/twbs/bootstrap/blob/v4-dev/scss/_buttons.scss#L159):
```css
.btn-block + .btn-block {
  margin-top: $btn-block-spacing-y;
}
```

I would force to create all the combination for the different responsive classes.
Unless we decide to get rid of it and rely on [spacing utilities](https://v4-alpha.getbootstrap.com/utilities/spacing/) instead

**Comment by vanduynslagerp** [comment]
For information this is what the scss would look like to handle the responsive `btn-{breakpoint}-block` and the spacing. Nasty...

```scss
@function breakpoint-infix-up($name, $breakpoints: $grid-breakpoints) {
  @return if(breakpoint-max($name, $breakpoints) == null, "", "-#{$name}");
}

@each $breakpoint in map-keys($grid-breakpoints) {
  @include media-breakpoint-down($breakpoint) {
    $infix: breakpoint-infix-up($breakpoint, $grid-breakpoints);

    .btn#{$infix}-block {
      display: block;
      width: 100%;
      @each $breakpoint2 in map-keys($grid-breakpoints) {
        $infix2: breakpoint-infix-up($breakpoint2, $grid-breakpoints);

        + .btn#{$infix2}-block {
          margin-top: $btn-block-spacing-y;
        }

        .btn#{$infix2}-block + & {
          margin-top: $btn-block-spacing-y;
        }
      }
    }
  }
}
```

**Comment by vanduynslagerp** [comment]
I went ahead and made the PR #21720.
It generates the responsive `btn-*-block` but not the vertical spacing.
The docs now mention to use the [spacing utilities](https://v4-alpha.getbootstrap.com/utilities/spacing/) instead.

**Comment by guylepage3** [comment]
Definitely would like to see responsive `btn-reset` or something of the sort.

**Comment by beckbailey** [comment]
Just add white-space: normal; to your .btn class.

**Comment by tomsommer** [comment]
It's a shame #21720 was closed. This feature really makes sense for buttons, especially since there is responsive utility for most other things.

It should be possible to find a non-breaking middle-ground based on #21720?

**Comment by MartijnCuppens** [comment]
We can indeed achieve this with utility classes these days, like @pvdlg mentioned.
